src/context/StateProvider.js

```diff
@@ -9,24 +17,33 @@ const StateProvider = ({ children }) => {
   const initialProductState = {
     cart: [],
     wishlist: [],
-    productList: [],
+    productsList: [],
   };
 
-  const [{ cart, wishlist, productList }, productDispatch] = useReducer(
-    productReducer,
+  const [{ cart, wishlist, productsList }, productsDispatch] = useReducer(
+    productsReducer,
     initialProductState
   );
 
+  useEffect(() => {
+    axios.get("/api/products").then((response) => {
```

Comment:
do api calls in try catch it will help in debugging if you get any errors

Reply:
I will do it. Thank you for the review!